src/dom.js

```diff
@@ -375,6 +375,21 @@ export function childElementByTag(parent, tagName) {
 
 
 /**
+ * A wrapper around native querySelectorAll to use :scope when possible.
+ * @param {!Element} parent
+ * @param {string} selector
+ * @returns {!Array.<!Element>}
+ */
+export function querySelectorAll(parent, selector) {
```

Comment:
No not a polyfill, @dvoytenko [suggested](https://github.com/ampproject/amphtml/pull/3734#discussion_r68316264) instead of making the call to the native querySelectorAll is to try to move it to a helper method that would use `:scope` when available.